src/service/logs/ingest.rs

```diff
@@ -208,38 +207,17 @@ pub async fn ingest(
         // end row based transformation
 
         // JSON Flattening
-        let item = flatten::flatten_with_level(item, cfg.limit.ingest_flatten_level)?;
-
-        // Start re-routing if exists
-        if let Some(routings) = stream_routing_map.get(&routed_stream_name) {
-            if !routings.is_empty() {
-                for route in routings {
-                    let mut is_routed = true;
-                    let val = &route.routing;
-                    for q_condition in val.iter() {
-                        if !q_condition.evaluate(item.as_object().unwrap()).await {
```

Comment:
This old code evaluated routing conditions on the data item that...
- had pre-flattening-functions applied to it,
- and was flattened,
- but did **not** yet have post-flattening functions applied to it

In contrast, the new code evaluates routing conditions on the data **after** post-flattening-functions have been applied.